contrib/registry/consul/client.go

```diff
@@ -103,26 +116,38 @@ func (c *Client) Register(_ context.Context, svc *registry.ServiceInstance, enab
 		asr.Address = host
 		asr.Port = int(port)
 	}
-	if enableHealthCheck {
-		for _, address := range checkAddresses {
-			asr.Checks = append(asr.Checks, &api.AgentServiceCheck{
-				TCP:                            address,
-				Interval:                       "20s",
-				DeregisterCriticalServiceAfter: "70s",
-			})
+	asrList := c.filter([]*api.AgentServiceRegistration{asr})
```

Comment:
Here, I feel that it should not be registered for more than one time. 
If it is registered for more than one time, there will be a problem of cancellation of registration.

Reply:
You are right, let me think how to fix it.Here I hope to register different endpoints as different instances and mark them with tags.

Reply:
Now that the PR of the Listener option has been merged, multiple protocols can reuse ports, and multiple ports can be registered as multiple instances with App. So only one is really registered here.